### PR TITLE
[BIA-483] Creates the StudentAssessmentFact View.

### PR DESCRIPTION
**Pull request by DavidJGapCR**
### Testing
The testing for these tickets (where we create new views or modify existing ones), have a few steps.
1. The first steps would be to make sure all tests run successfully. For this use the Test Explorer of Visual Studio. 
Execute all the Common section and Operation section. For the Dimensions section execute at least the new ones created, associates to the new view. (screenshot attached) It's not a bad idea to execute them all, but it takes some time.

2. Make sure you are able to install AMT on the different data standards. Ideally you would test on all of them: 3.1, 3.2, 3.3, 3.4, 3.5. 
If you go to the build of the Console project, you can open a command prompt and execute something like this to install:
`dotnet EdFi.AnalyticsMiddleTier.Console.dll --connectionString "Data Source=IT-CR-0868;Initial Catalog=Ed-Fi-Glendale-3.1.0;Integrated Security=SSPI;" --options Indexes Qews Rls Ews Chrab Asmt`
And something like this to uninstall:
`dotnet EdFi.AnalyticsMiddleTier.Console.dll --connectionString "Data Source=IT-CR-0868;Initial Catalog=Ed-Fi-Glendale-3.1.0;Integrated Security=SSPI;" --uninstall`

3. Review the view design. for example: Is it returning the fields it should based on the ticket description? Are the field names correct? Are the joins properly defined based on the foreign keys on the tables? Etc. Etc.

4. When we create the tests, the idea is to cover all possible scenarios of the possible results. You can do that by just reviewing the test scenarios. (_\src\EdFi.AnalyticsMiddleTier.Tests\TestCases\StudentAssessmentFact\_). And something we do as well, is to comment out sections of the view.
For example, open the new view, and comment out line 29 (`AND StudentAssessment.Namespace = Assessment.Namespace`). If you run the dimension tests for this view, at least one test must fail. 
You can do this for all the joins with more than 1 condition.
You can do this with the **where** statement at the bottom of the view. 
Etc.

### CODE REVIEW

**Review by kurtjohnson-HS — COMMENTED** [review]
I tested the install against v3.1, v3.2, v3.3, v3.4, and v5.0 for MSSQL. The Postgres Console migration fails - as you already know.

I did receive 22 errors after commenting out the namespace in the joins - more than 1 error as expected. 
image

**Review by kurtjohnson-HS — CHANGES_REQUESTED** [review]
We still need to resolve the question of whether or not to change the edfi.Assessment  table join to the asmt_AssessmentDim view to match the work item.

File: src/EdFi.AnalyticsMiddleTier.DataStandard31/Asmt/MSSQL/0001-View-StudentAssessmentFact-Create.sql [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        School.SchoolId AS SchoolKey,
+        CONVERT(varchar, StudentAssessment.AdministrationDate, 112) as AdministrationDate,
+        StudentAssessmentScoreResult.Result AS StudentScore,
+        ResultDatatypeTypeDescriptorDist.Description AS ResultDataType,
+        AssessmentReportingMethodDescriptorDist.Description AS ReportingMethod,
+        PerformanceLevelDescriptorDist.Description AS PerformanceResult
+    FROM
+        edfi.StudentAssessment
+    INNER JOIN
+        edfi.Assessment
```
[the remark this answers is not in the document]
Comment (DavidJGapCR) [reply src/EdFi.AnalyticsMiddleTier.DataStandard31/Asmt/MSSQL/0001-View-StudentAssessmentFact-Create.sql:27]: I don't think it says we have to use the AssessmentDim view. I think what it says is that this field is going to be used as a key to associate StudentAssessmentFact with AssessmentDim. Actually, if you see the other fields: StudentSchoolKey and SchoolKey it says basically the same thing, but we don't use these views.
We usually don't join views with other views. If I remember correctly there is just one case where we did that. I don't remember where exactly.

**Comment by DavidJGapCR** [comment]
> I tested the install against v3.1, v3.2, v3.3, v3.4, and v5.0 for MSSQL. The Postgres Console migration fails - as you already know.
> 
> I did receive 22 errors after commenting out the namespace in the joins - more than 1 error as expected.
> image

Just to confirm we are on the same page.
You can comment out not only the namespace condition in the joins but any condition in a join with more than 1 condition.
For example, lines:
28,29,44,45,46,47,50,51,52,53,71,74
Basically, where there is an AND or OR in a join, you can comment out one side or the other.
Let me know if this makes sense.

**Comment by kurtjohnson-HS** [comment]
> Just to confirm we are on the same page.
> You can comment out not only the namespace condition in the joins but any condition in a join with more than 1 condition.
> For example, lines:
> 28,29,44,45,46,47,50,51,52,53,71,74
> Basically, where there is an AND or OR in a join, you can comment out one side or the other.
> Let me know if this makes sense.

That was what I did, so we are on the same page. I went through each one and verified that testing threw exceptions when the condition was commented out.

**Review by kurtjohnson-HS — APPROVED** [review]

**Pull request merged**